ccmlib/node.py

```diff
@@ -680,16 +680,20 @@ def stop(self, wait=True, wait_other_notice=False, gently=True):
         else:
             return False
 
-    def wait_for_compactions(self):
+    def wait_for_compactions(self, timeout=600):
         """
         Wait for all compactions to finish on this node.
         """
         pattern = re.compile("pending tasks: 0")
-        while True:
+        elapsed = 0
+        while elapsed < timeout:
             output, err = self.nodetool("compactionstats", capture_output=True)
             if pattern.search(output):
-                break
-            time.sleep(10)
+                return
+            elapsed = elapsed + 1
+            time.sleep(1)
+        raise TimeoutError(time.strftime("%d %b %Y %H:%M:%S", time.gmtime()) + " [" + self.name + "] Compactions did not finish in {} seconds".format(timeout))
```

Comment:
Could you use `.format()` to build this entire string, and not do any concatenation?